Code/GraphMol/MolPickler.cpp

```diff
@@ -473,6 +473,7 @@ void pickleQuery(std::ostream &ss, const Query<int, T const *, true> *query) {
         streamWrite(ss, MolPickler::QUERY_VALUE, std::get<2>(v));
         streamWriteProp(ss, std::get<1>(v),
                         MolPickler::getCustomPropHandlers());
+        std::get<1>(v).cleanup();
```

Comment:
This is the "real" fix here.

Reply:
I think it's actually redundant since the destructor of the `PairHolder` calls `cleanup_rdvalue()`.